config/fi_provider.m4

```diff
@@ -192,3 +193,56 @@ AC_DEFUN([_FI_PROVIDER_INVOKE],[
 		[])]
 	    )
 ])
+
+dnl
+dnl Internal; should not be called from provder .m4 scripts.
+dnl Helper macro that appends each providers build flags for pkg-config
+dnl
+dnl Arguments:
+dnl
+dnl $1: name of the provider
+dnl
+AC_DEFUN([_FI_PROVIDER_PC_ADD],[
+	 FI_PC_CFLAGS="$FI_PC_CFLAGS $$1_CFLAGS $$1_CPPFLAGS"
+	 FI_PC_PRIVATE_LIBS="$FI_PC_PRIVATE_LIBS $$1_LIBS"
+	 # usnic with nl needs special handling (should be added to usnic_LIBS?)
+	 AS_IF([test "$1" == "usnic"], [
```

Comment:
I looked into this a bit.  It looks like the sockets, verbs, and usnic providers all have additional libraries outside of `<provider_name>_LIBS`, and won't be captured here.

I can file a PR to make the usnic provider condense down to just `$usnic_LIBS` (i.e., `$$1_LIBS`).  Once that's in place, this special handling for the usnic provider should go away.

@jithinjosepkl @shantonu can you see if sockets can condense down to just `$sockets_LIBS`?
@ddurnov @evgeny-leksikov can you see if verbs can condense down to just `$verbs_LIBS`?